ably-ios/ARTRealtime.m

```diff
@@ -758,7 +768,11 @@ - (void)realtimeTransportClosed:(id<ARTRealtimeTransport>)transport {
 }
 
 - (void)realtimeTransportDisconnected:(id<ARTRealtimeTransport>)transport {
-    [self transition:ARTRealtimeDisconnected];
+    if (self.connection.state == ARTRealtimeClosing) {
+        [self transition:ARTRealtimeClosed];
```

Comment:
But this is a disconnection. If there is a disconnection, and it was closing, the CLOSED action obviously can't arrive, but the client should act as if it happened. That's what I understand from [RTN12c](http://docs.ably.io/client-lib-development-guide/features/#RTN12c).